dbms/src/Storages/MergeTree/MergeTreeDataMergerMutator.cpp

```diff
@@ -211,7 +217,10 @@ bool MergeTreeDataMergerMutator::selectPartsToMerge(
         merge_settings.base = 1;
 
     /// NOTE Could allow selection of different merge strategy.
-    merge_selector = std::make_unique<SimpleMergeSelector>(merge_settings);
+    if (has_part_with_expired_ttl)
```

Comment:
What if we get parts with TTL and their values are expires each second? Seems like we will merge only, and touch other parts which may require "normal merge".